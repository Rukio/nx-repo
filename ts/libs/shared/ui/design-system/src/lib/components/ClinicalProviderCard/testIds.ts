const CLINICAL_PROVIDER_CARD_TEST_IDS = {
  // test id strings will be prefixed by the testIdPrefix prop (ex. xyz-clinical-provider-card)
  CLINICAL_PROVIDER_CARD: 'clinical-provider-card',
  PROVIDER_NAME: 'provider-name',
  PROVIDER_ADDRESS_LINE_1: 'provider-address-line1',
  PROVIDER_CITY_STATE: 'provider-city-state',
  PROVIDER_PHONE_CONTAINER: 'provider-phone-container',
  PROVIDER_PHONE: 'provider-phone',
  PROVIDER_FAX_CONTAINER: 'provider-fax-container',
  PROVIDER_FAX: 'provider-fax',
  PROVIDER_SELECT_BUTTON: 'provider-select-button',
};

export default CLINICAL_PROVIDER_CARD_TEST_IDS;
