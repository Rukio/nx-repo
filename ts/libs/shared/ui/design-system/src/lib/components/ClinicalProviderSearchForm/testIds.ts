const CLINICAL_PROVIDER_SEARCH_FORM_TEST_IDS = {
  FORM_SELECT_BUTTON: 'form-select-button', // with form name suffix, ex. form-select-button-practice
  FIRST_NAME_INPUT: 'first-name-input',
  LAST_NAME_INPUT: 'last-name-input',
  PRACTICE_NAME_INPUT: 'practice-name-input',
  PHARMACY_NAME_INPUT: 'pharmacy-name-input',
  RANGE_IN_MILES_INPUT: 'range-in-miles-input',
  DISTANCE_VALUE: 'distance-value', // with distance as suffix, ex. distance-value-25
  ZIP_CODE_INPUT: 'zip-code-input',
  PHONE_INPUT: 'phone-input',
  INPUT_ERROR: 'error', // with field test id as prefix, ex. last-name-input-error
  SEARCH_BUTTON: 'search-button',
};

export default CLINICAL_PROVIDER_SEARCH_FORM_TEST_IDS;
