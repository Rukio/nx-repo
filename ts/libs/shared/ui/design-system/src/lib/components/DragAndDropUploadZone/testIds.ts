export const DRAG_AND_DROP_TEST_IDS = {
  getDragAndDropUploadZone: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone`,
  getDragAndDropUploadZoneText: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-text`,
  getDragAndDropZoneUploadErrorText: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-error-text`,
  getDragAndDropZoneUploadInput: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-input`,
};
