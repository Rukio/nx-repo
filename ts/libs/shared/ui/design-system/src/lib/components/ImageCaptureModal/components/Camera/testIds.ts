export const CAMERA_TEST_IDS = {
  getCameraTopText: (testIdPrefix: string) => `${testIdPrefix}-camera-top-text`,
  getCameraBox: (testIdPrefix: string) => `${testIdPrefix}-camera-box`,
  getCameraCanvasBox: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-canvas-box`,
  getCameraBottomText: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-bottom-text`,
  getCameraLoadingIndicator: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-loading-indicator`,
};
