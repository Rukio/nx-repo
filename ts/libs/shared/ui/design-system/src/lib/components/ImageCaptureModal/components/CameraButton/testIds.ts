export const CAMERA_BUTTON_TEST_IDS = {
  getCaptureButton: (testIdPrefix: string) =>
    `${testIdPrefix}-photo-modal-capture-photo-button`,
};
