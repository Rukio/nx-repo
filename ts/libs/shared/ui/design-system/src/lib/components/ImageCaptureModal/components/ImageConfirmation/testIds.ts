export const IMAGE_CONFIRMATION_TEST_IDS = {
  getImageConfirmationTopText: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-top-text`,
  getImageConfirmationImage: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-image`,
  getImageConfirmationImagePlaceholder: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-image-placeholder`,
  getImageConfirmationRetakeButton: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-retake-button`,
  getImageConfirmationButton: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-button`,
};
