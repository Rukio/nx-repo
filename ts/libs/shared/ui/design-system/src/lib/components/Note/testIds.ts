const notePrefixText = 'note';
const noteContainerPrefixText = 'note-action';

export const NOTE_TEST_IDS = {
  getNoteByTestId: (noteId: number) => `${notePrefixText}-${noteId}`,
  getNoteActionByTestId: (noteId: number) =>
    `${noteContainerPrefixText}-${noteId}`,
  INITIALS: 'note-initials',
  DISPLAY_NAME: 'note-display-name',
  SECONDARY_LABEL: 'note-secondary-label',
  TEXT: 'note-text',
  MORE_BUTTON: 'note-more-button',
  EDIT_MENU_ITEM: 'note-edit-menu-item',
  TOGGLE_FEATURED_MENU_ITEM: 'note-toggle-featured-menu-item',
  FEATURED_NOTE_CHIP: 'note-featured-note-chip',
  DELETE_MENU_ITEM: 'note-delete-menu-item',
  DELETED_ALERT: 'note-deleted-alert',
  UNDO_DELETE_BUTTON: 'note-undo-delete-button',
  EDIT_INPUT: 'note-edit-input',
  EDIT_CANCEL_BUTTON: 'note-edit-cancel-button',
  EDIT_SAVE_CHANGES_BUTTON: 'note-edit-save-changes-button',
};
