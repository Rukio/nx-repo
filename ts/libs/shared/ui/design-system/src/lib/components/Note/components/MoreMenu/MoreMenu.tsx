import { FC, useRef, useState } from 'react';
import { MoreHorizIcon } from '../../../../icons/MoreHoriz';
import Typography from '../../../Typography';
import { NOTE_TEST_IDS } from '../../testIds';
import { IconButton, Menu, MenuItem } from '../../../..';

export type MoreMenuProps = {
  isFeaturedNoteEnabled?: boolean;
  featured?: boolean;
  isEditingEnabled?: boolean;
  isDeletingEnabled?: boolean;
  onFeaturedClick?: () => void;
  onDeleteClick?: () => void;
  onEditClick?: () => void;
};

const MoreMenu: FC<MoreMenuProps> = ({
  isFeaturedNoteEnabled = true,
  featured,
  isEditingEnabled = true,
  isDeletingEnabled = true,
  onFeaturedClick,
  onDeleteClick,
  onEditClick,
}) => {
  const [open, setOpen] = useState(false);
  const buttonAnchorRef = useRef<HTMLDivElement>(null);

  const toggleMenu = () => {
    setOpen((prev) => !prev);
  };

  const featureMenuItemLabel = featured
    ? 'Unfeature Note'
    : 'Make Featured Note';

  const showMenuControlButton =
    isEditingEnabled || isDeletingEnabled || isFeaturedNoteEnabled;

  if (!showMenuControlButton) {
    return null;
  }

  const handleFeatureClick = () => {
    toggleMenu();
    onFeaturedClick?.();
  };

  return (
    <>
      <div ref={buttonAnchorRef}>
        <IconButton
          size="small"
          onClick={toggleMenu}
          data-testid={NOTE_TEST_IDS.MORE_BUTTON}
        >
          <MoreHorizIcon fontSize="small" />
        </IconButton>
      </div>
      <Menu
        open={open}
        anchorEl={buttonAnchorRef.current}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        onClose={toggleMenu}
      >
        {isEditingEnabled && (
          <MenuItem
            onClick={onEditClick}
            data-testid={NOTE_TEST_IDS.EDIT_MENU_ITEM}
          >
            <Typography variant="body2">Edit</Typography>
          </MenuItem>
        )}
        {isFeaturedNoteEnabled && (
          <MenuItem
            onClick={handleFeatureClick}
            data-testid={NOTE_TEST_IDS.TOGGLE_FEATURED_MENU_ITEM}
          >
            <Typography variant="body2">{featureMenuItemLabel}</Typography>
          </MenuItem>
        )}
        {isDeletingEnabled && (
          <MenuItem
            onClick={onDeleteClick}
            data-testid={NOTE_TEST_IDS.DELETE_MENU_ITEM}
          >
            <Typography color="error" variant="body2">
              Delete
            </Typography>
          </MenuItem>
        )}
      </Menu>
    </>
  );
};

export default MoreMenu;
