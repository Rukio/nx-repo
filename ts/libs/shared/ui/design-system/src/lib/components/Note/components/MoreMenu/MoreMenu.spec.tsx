import { render, userEvent, screen, waitFor } from '../../../../../test-utils';
import { NOTE_TEST_IDS } from '../../testIds';
import MoreMenu, { MoreMenuProps } from './MoreMenu';

const mockDefaultProps: MoreMenuProps = {
  onFeaturedClick: jest.fn(),
  onEditClick: jest.fn(),
  onDeleteClick: jest.fn(),
};

const setup = (props: Partial<MoreMenuProps> = {}) => ({
  ...render(<MoreMenu {...mockDefaultProps} {...props} />),
  user: userEvent.setup(),
});

describe('MoreMenu', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render with closed menu correctly', async () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call onToggleFeatured on menu item click if note is not featured', async () => {
    const { user } = setup();

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const toggleFeaturedMenuItem = await screen.findByTestId(
      NOTE_TEST_IDS.TOGGLE_FEATURED_MENU_ITEM
    );
    expect(toggleFeaturedMenuItem).toBeVisible();
    expect(toggleFeaturedMenuItem).toBeEnabled();
    expect(toggleFeaturedMenuItem).toHaveTextContent('Make Featured Note');

    await user.click(toggleFeaturedMenuItem);

    await waitFor(() => {
      expect(toggleFeaturedMenuItem).not.toBeVisible();
    });

    await waitFor(() => {
      expect(mockDefaultProps.onFeaturedClick).toBeCalled();
    });
  });

  test('should call onToggleFeatured on menu item click if note is featured', async () => {
    const { user } = setup({ featured: true });

    const moreButton = screen.getByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const toggleFeaturedMenuItem = await screen.findByTestId(
      NOTE_TEST_IDS.TOGGLE_FEATURED_MENU_ITEM
    );
    expect(toggleFeaturedMenuItem).toBeVisible();
    expect(toggleFeaturedMenuItem).toBeEnabled();
    expect(toggleFeaturedMenuItem).toHaveTextContent('Unfeature Note');

    await user.click(toggleFeaturedMenuItem);

    await waitFor(() => {
      expect(mockDefaultProps.onFeaturedClick).toBeCalled();
    });
  });

  test('should not show "Make featured" option if isFeaturedNoteEnabled is falsy', async () => {
    const { user } = setup({ isFeaturedNoteEnabled: false });

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const toggleFeaturedMenuItem = screen.queryByTestId(
      NOTE_TEST_IDS.TOGGLE_FEATURED_MENU_ITEM
    );
    expect(toggleFeaturedMenuItem).not.toBeInTheDocument();
  });

  test('should not show "Edit" option if isEditingEnabled is falsy', async () => {
    const { user } = setup({ isEditingEnabled: false });

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const editMenuItem = screen.queryByTestId(NOTE_TEST_IDS.EDIT_MENU_ITEM);
    expect(editMenuItem).not.toBeInTheDocument();
  });

  test('should call onEditClick on menu item click', async () => {
    const { user } = setup();

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const editMenuItem = await screen.findByTestId(
      NOTE_TEST_IDS.EDIT_MENU_ITEM
    );
    expect(editMenuItem).toBeVisible();
    expect(editMenuItem).toBeEnabled();

    await user.click(editMenuItem);

    await waitFor(() => {
      expect(mockDefaultProps.onEditClick).toBeCalled();
    });
  });

  test('should not show "Delete" option if isDeletingEnabled is falsy', async () => {
    const { user } = setup({ isDeletingEnabled: false });

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const deleteMenuItem = screen.queryByTestId(NOTE_TEST_IDS.DELETE_MENU_ITEM);
    expect(deleteMenuItem).not.toBeInTheDocument();
  });

  test('should call onDeleteClick on menu item click', async () => {
    const { user } = setup();

    const moreButton = await screen.findByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).toBeVisible();
    expect(moreButton).toBeEnabled();

    await user.click(moreButton);

    const deleteMenuItem = await screen.findByTestId(
      NOTE_TEST_IDS.DELETE_MENU_ITEM
    );
    expect(deleteMenuItem).toBeVisible();
    expect(deleteMenuItem).toBeEnabled();

    await user.click(deleteMenuItem);

    await waitFor(() => {
      expect(mockDefaultProps.onDeleteClick).toBeCalled();
    });
  });

  test('should not show more button if isDeletingEnabled, isEditingEnabled, isFeaturedNoteEnabled options are falsy', () => {
    setup({
      isDeletingEnabled: false,
      isEditingEnabled: false,
      isFeaturedNoteEnabled: false,
    });

    const moreButton = screen.queryByTestId(NOTE_TEST_IDS.MORE_BUTTON);
    expect(moreButton).not.toBeInTheDocument();
  });
});
