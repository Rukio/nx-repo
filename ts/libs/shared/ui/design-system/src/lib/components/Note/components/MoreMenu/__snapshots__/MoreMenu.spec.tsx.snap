// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreMenu should render with closed menu correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(17, 27, 34, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 5px;
  font-size: 1.125rem;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(17, 27, 34, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(17, 27, 34, 0.26);
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall emotion-0"
      data-testid="note-more-button"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-1"
        data-testid="MoreHorizIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
  </div>
</DocumentFragment>
`;
